common/djangoapps/status/models.py

```diff
@@ -0,0 +1,53 @@
+"""
+Store status messages in the database.
+"""
+
+from django.db import models
+from django.contrib import admin
+
+from xmodule_django.models import CourseKeyField
+
+from config_models.models import ConfigurationModel
+from config_models.admin import ConfigurationModelAdmin
+
+
+class GlobalStatusMessage(ConfigurationModel):
+    """
+    Model that represents the current status message.
+    """
+    message = models.TextField(blank=True, null=True)
+
+    def full_message(self, course_key):
+        """ Returns the full status message, including any course-specific status messages. """
+        msg = self.message
+        if course_key:
+            try:
+                course_message = self.coursemessage_set.get(course_key=course_key)
+                # Don't add the message if course_message is blank.
+                if course_message:
+                    msg = u"{} <br /> {}".format(msg, course_message.message)
+            except CourseMessage.DoesNotExist:
+                # We don't have a course-specific message, so pass.
+                pass
+        return msg
+
+    def __unicode__(self):
+        return "{} - {} - {}".format(self.change_date, self.enabled, self.message)
+
+
+class CourseMessage(models.Model):
+    """
+    Model that allows the user to specify messages for individual courses.
+
+    This is not a ConfigurationModel because the managing of the foreign key wouldn't work the way we'd like.
```

Comment:
So I can update this explanation, but the real problem is a lot simpler than I originally thought. We can only have one current `ConfigurationModel` at a time, which makes it impossible/confusing for us to set different messages for different courses.